utilities/write_batch_with_index/write_batch_with_index.cc

```diff
@@ -310,18 +319,21 @@ class BaseDeltaIterator : public Iterator {
           }
         } else {
           current_at_base_ = true;
-          return;
+          break;
         }
       }
     }
 
-    AssertInvariants();
+    current_over_upper_bound_ = IsOverUpperBound();
```

Comment:
Sorry I rerethink about it. Here it still iterates through the tombstones in the write batch, which can be solved by doing the upper bound check inside the while loop.

Reply:
Do we need to do the comparison again if it is broken from the loop?

Reply:
depends on where we broke from the loop, if we broke because `current_over_upper_bound_` is true, then it's unnecessary, but if we broke for other reasons, then the upper bound may have been exceeded but `current_over_upper_bound_` is not updated yet, which is why I call `IsOverUpperBound()` outside the loop

Reply:
I think there is some issue with breaking early from the loop here. 
Suppose we start with keys of {a,b,c,d,e,f,g,h,i,j} and after some deletes, the valid key space becomes {a,c,d,e}, and iterate_upper_bound is set at "f".
If we do `SeekForPrev("g")`, we will first see 'f' and `IsOverUpperBound()` will determine we are out-of-bound and invalidate the iterator, but if we keep going we will find out the correct value is 'e', which not out-of-bound. So I suppose for correctness we still need to iterate through all tombstones here?